### ISSUE TITLE
Cross references are not translated

### ISSUE THREAD

**Issue by dtenbrinck** [executablebooks/sphinx-proof#125]
### Describe the bug

**context**
When I compile my markdown files to HTML using sphinx_proof with language settings to German, the titles of most of the proof boxes are translated correctly (some are missing) but the cross references to these boxes stay in English language.

**expectation**
I expect that the titles in cross references are also translated according to the language settings.

**problem**
This gives inconsistent results in books that are not written in English language.

![sphinx_proof_bug](https://github.com/user-attachments/assets/765f35f7-2ff2-45d8-9caf-8d407a2ccfee)

### Reproduce the bug

1. Add the following to your `_config.yml`:

```yaml
sphinx:
  extra_extensions:
    - sphinx_proof
  config:
    language: de
```

2. Compile book to HTML with `jupyter-book build .`

3. Observe build result in folder `_build/html/`

### List your environment

```
Version of sphinx-proof: 0.2.0

jupyter-book --version
Jupyter Book      : 1.0.3
External ToC      : 1.0.1
MyST-Parser       : 2.0.0
MyST-NB           : 1.1.2
Sphinx Book Theme : 1.1.3
Jupyter-Cache     : 1.0.0
NbClient          : 0.10.0

python --version
Python 3.12.7

Operating System: ArchLinux
```

**Comment by MPenaR** [comment]
I have the same setting but with "language: es" expecting to get the spanish but they are in english (both titles and references). 

Is this feature only available in german?

**Comment by dtenbrinck** [comment]
Probably you are using the pip installed package of `sphinx-proof`, which unfortunately doesn't include the directory `sphinx_proof/translations/jsons` with the needed translations (you can verify on [PyPI](https://pypi.org/project/sphinx-proof/#files)).
To make this work you need to clone this repository and add Spanish translations in some of the json files as follows:

1. Clone the repository, e.g., `git clone https://github.com/executablebooks/sphinx-proof.git ~/`
2. Add translations in the files in `~/sphinx-proof/sphinx_proof/translations/jsons/` (see [my commit](https://github.com/executablebooks/sphinx-proof/pull/126/commits/b1eedfa2cd695c2e616b19dc7beca4ed939bb48f) for example)
3. Generate the translation binaries via `python ~/sphinx-proof/sphinx_proof/translations/_convert.py`
4. Link the package into your virtual environment, e.g., via `ln -s ~/sphinx-proof/sphinx_proof ~/my_book_project/.venv/lib/python3.12/site-packages/sphinx_proof`
5. Configure your sphinx settings in the file `~/my_book_project/_config.yml` with:

```yaml
sphinx:
  extra_extensions:
    - sphinx_proof
  config:
    language: es
```

It would be great if you could create a pull request for your contribution of Spanish translations. :)

**Comment by MPenaR** [comment]
Thank you for the detailed explanation, that way not only will be able to have spanish translations but I also learn XD

I'll try to do it this evening

**Comment by MPenaR** [comment]
When I run the `python _convert.py` it complains about not finding the file or directory `msgfmt`. Do you know which is the minimal enviroment I need?

**Comment by MPenaR** [comment]
For someone in the future, in debian the utility `msgfmt` is installed with the package `gettext`

**Comment by MPenaR** [comment]
> It would be great if you could create a pull request for your contribution of Spanish translations. :)

I think I did it correctly. It's in pull request #127

**Comment by dtenbrinck** [comment]
Great work! I hope our pull requests will be merged soon into `sphinx-proof`. :+1:

**Comment by MPenaR** [comment]
I've just realised that my changes only affect the titles of the directives, but not the cross references. But if I understood it correctly, once my code and your code get merged the spanish cross references will also work. 

Am I right?

**Comment by dtenbrinck** [comment]
Yes, my changes should work for any translation. You can test it by merging [my commit](https://github.com/executablebooks/sphinx-proof/pull/126/commits/3b421e63483a30c73c16838bca99ec437b14a2b7) into your code.

**Comment by MPenaR** [comment]
At the beggining of a "proof" directive it is still read _proof_ independently of the language used:
![imagen](https://github.com/user-attachments/assets/fbe90105-b729-4b44-b8b3-d1697f315af3)


Is this a feature or a bug?

**Comment by dtenbrinck** [comment]
I think this is fixed in [this open pull request](https://github.com/executablebooks/sphinx-proof/pull/119).

**Comment by MPenaR** [comment]
I see, thank you very much
